tasks/deploy.js

```diff
@@ -0,0 +1,955 @@
+const { ratioToSqrtPriceX96, sqrtPriceX96ToPrice, } = require(`${__dirname}../../test/lib/sqrtPriceUtils.js`);
+
+const testnets = ['goerli', 'hardhat'];
+const moduleIds = {
+    // Public single-proxy modules
+    INSTALLER: 1,
+    MARKETS: 2,
+    LIQUIDATION: 3,
+    GOVERNANCE: 4,
+    EXEC: 5,
+    SWAP: 6,
+
+    // Public multi-proxy modules
+    ETOKEN: 500000,
+    DTOKEN: 500001,
+
+    // Internal modules
+    RISK_MANAGER: 1000000,
+
+    IRM_DEFAULT: 2000000,
+    IRM_ZERO: 2000001,
+    IRM_FIXED: 2000002,
+    IRM_LINEAR: 2000100,
+};
+
+const contractNames = [
+    // Core
+
+    'Euler',
+
+    // Modules
+
+    'Installer',
+    'Markets',
+    'Liquidation',
+    'Governance',
+    'Exec',
+    'Swap',
+    'EToken',
+    'DToken',
+
+    // Internal modules
+
+    'RiskManager',
+    'IRMDefault',
+    'IRMZero',
+    'IRMFixed',
+    'IRMLinear',
+
+    // Adaptors
+
+    'FlashLoan',
+
+    // Liquidity Mining
+
+    'EulStakes',
+    'EulDistributor',
+
+    // Testing
+
+    'TestERC20',
+    'MockUniswapV3Factory',
+    'EulerGeneralView',
+    'EulerSimpleLens',
+    'InvariantChecker',
+    'FlashLoanNativeTest',
+    'FlashLoanAdaptorTest',
+    'SimpleUniswapPeriphery',
+    'TestModule',
+    'MockAggregatorProxy'
+];
+
+const defaultUniswapFee = 3000;
+const PRICINGTYPE__UNISWAP3_TWAP = 2;
+const PRICINGTYPE__CHAINLINK = 4;
+
+task("testTokenFaucet:deploy")
+    .addPositionalParam("threshold", "A number specifying the maximum amount of tokens any user can hold at any point in time")
+    .setAction(async (args) => {
+        const tokenFaucetFactory = await ethers.getContractFactory('TestERC20TokenFaucet');
+
+        let tx = await tokenFaucetFactory.deploy(args.threshold);
+        console.log(`Transaction: ${tx.deployTransaction.hash}`);
+
+        let result = await tx.deployed();
+        console.log(`Contract: ${result.address}`);
+    });
+
+task("chainlinkOracle:deploy")
+    .addOptionalParam("decimals", "decimals", "18")
+    .setAction(async (args) => {
+        const priceOracleFactory = await ethers.getContractFactory('MockAggregatorProxy');
+        
+        let tx = await priceOracleFactory.deploy(parseInt(args.decimals));
+        console.log(`Transaction: ${tx.deployTransaction.hash}`);
+
+        let result = await tx.deployed();
+        console.log(`Contract: ${result.address}`);
+    });
+
+task("chainlinkOracle:setValidAnswer")
+    .addPositionalParam("impl", "Price oracle smart contract address")
+    .addPositionalParam("price")
+    .setAction(async (args) => {
+        const oracle = await ethers.getContractAt('MockAggregatorProxy', args.impl);
+        
+        let tx = await oracle.mockSetValidAnswer(args.price);
+        console.log(`Transaction: ${tx.hash}`);
+
+        let result = await tx.wait();
+        console.log(`Status: ${result.status}`);
+    });
+
+task("deploy:update-network", "Update the current state of Euler smart contracts and markets")
```

Comment:
This looks like a lot of code duplication with eTestLib. Wouldn't it be possible to refactor the lib so that code could be reused?

Reply:
That's right. I can move some code over to the lib to make the hardhat tasks lighter and test it. There is some addition @dglowinski